internal/service/networkcontainer/resource_network_container_test.go

```diff
@@ -35,31 +35,11 @@ func TestAccNetworkContainer_basicAWS(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config: configBasic(projectID, cidrBlock, constant.AWS, "US_EAST_1"),
-				Check: resource.ComposeTestCheckFunc(
-					checkExists(resourceName),
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "provider_name", constant.AWS),
-					resource.TestCheckResourceAttrSet(resourceName, "provisioned"),
-
-					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
-					resource.TestCheckResourceAttr(dataSourceName, "provider_name", constant.AWS),
-					resource.TestCheckResourceAttrSet(dataSourceName, "provisioned"),
-
-					resource.TestCheckResourceAttrWith(dataSourcePluralName, "results.#", acc.IntGreatThan(0)),
-					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.id"),
-					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.atlas_cidr_block"),
-					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provider_name"),
-					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.provisioned"),
-				),
+				Check:  resource.ComposeTestCheckFunc(commonChecks(constant.AWS)...),
```

Comment:
@EspenAlbert at then end i did the refactor you recommended following @AgustinBettati example in search_deployment.   a lot of duplication gone and clearer to read.  (still this resource might have more specific attribute checks, not only that they exist).    cc @marcosuma